src/ec_sslwrap.c

```diff
@@ -419,6 +419,7 @@ static void sslw_hook_handled(struct packet_object *po)
       memcpy(s->data, &po->L3.dst, sizeof(struct ip_addr));
```

Comment:
Well generally, I wonder why the IP is not copied to the session data and the session isn't updated when running on Linux.
Right now, I don't see why Linux is exempted from this step.
Do you have an idea @LocutusOfBorg ?

Reply:
@koeppea I think we should ask  @eaescob , the commit came from him
af9b88b1587aa3c74ac73a3f4e198abd00841727

But looks maybe some wrong commit rather than deliberate change (according to the commit msg)

Reply:
also we should after calling
sslw_create_session

do something like:
SAFE_FREE(s->data);
SAFE_FREE(s);

right?

Reply:
Let me dig this up from my memory storage, please wait :) Will take a look in a bit

Reply:
I don't think the check makes sense at this point. overall agree (don't know what I was thinking) but if the original intent was for this to only work on Linux then moving up the pre-compiler condition makes sense as a solution

Reply:
Right. Actually SSL interception is properly working on Linux only AFAICS. I've to test MacOS. Support might be limited, due to Apple's efforts to nail down the operating system as hard as they can afford and due to the lack of OpenSSL.

But in the car this should only happen when on Linux, then the pre-compiler condition would have to be inverted, or am I thinking wrong?

In any case, dedicating an own function for properly freeing related memory is a good idea.

Reply:
I'd include this for Mac and Linux